### PR TITLE
anycable-go: update head build command

### PR DESCRIPTION
**Pull request by palkan**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Master is now pointing to v1.0 development, which changed the way build vars are passed (since https://github.com/anycable/anycable-go/pull/89 has been merged).

### CODE REVIEW

**Comment by SMillerDev** [comment]
What happens if you pass both? And maybe it can use `*std_go_args`?

**Comment by palkan** [comment]
> What happens if you pass both? 

If we pass a version in a HEAD mode, it would replace the current version specified in the code, and the output of `anycable-go -v` would be something like `HEAD-df76321`—no sign of the corresponding major/minor/patch versions, which. IMO, is not good for the end user.

This change aims to solve this problem by generating a version of the following format for HEAD builds: `1.0.0-df76321`.

> And maybe it can use *std_go_args?

Done.

File: Formula/anycable-go.rb [hunk 1]
```diff
@@ -15,8 +15,16 @@ class AnycableGo < Formula
...
+    build_vars =
+      if build.head?
+        "github.com/anycable/anycable-go/utils.sha=#{version.commit}"
+      else
+        "main.version=#{version}"
+      end
+
+    system "go", "build", "-mod=vendor", "-ldflags",
+                          "-s -w -X #{build_vars}",
+                          *std_go_args,
```
Comment (bayandin) [inline Formula/anycable-go.rb:27]: I would tune it a bit, just to keep `-X` closer with `build_vars`, something like this:

```suggestion
    ldflags = %w[
      -s -w
    ]
    ldflags << if build.head?
      "-X github.com/anycable/anycable-go/utils.sha=#{version.commit}"
    else
      "-X main.version=#{version}"
    end

    system "go", "build", "-mod=vendor", "-ldflags", ldflags.join(" "), *std_go_args,
```

**Comment by bayandin** [comment]
Thanks @palkan!

**Pull request closed without merging**